src/main/drivers/flash_w25q128fv.c

```diff
@@ -221,23 +251,23 @@ static bool w25q128fv_isWritable(flashDevice_t *fdevice)
     return writable;
 }
 
-bool w25q128fv_hasTimedOut(flashDevice_t *fdevice)
+MMFLASH_CODE bool w25q128fv_hasTimedOut(flashDevice_t *fdevice)
 {
-    uint32_t now = HAL_GetTick();
-    if (cmp32(now, fdevice->timeoutAt) >= 0) {
+    uint32_t nowMs = microsISR() / 1000;
+    if (cmp32(nowMs, fdevice->timeoutAt) >= 0) {
```

Comment:
This is the problem from above - cmp32() works only  with 32bit overflow

Reply:
Ahh yes, suggestions on what to do given that the timeout/delay functions need to work without systick ticking up the millis counter via `SysTick_Handler` then?

Reply:
since there's no suggestions for a month, and the code has been running on H7RF targets for over a year with zero reported problems we can look to improve this in the future once a suggestion/solution presents itself.